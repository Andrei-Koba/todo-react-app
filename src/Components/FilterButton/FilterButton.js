import React, { useState } from 'react';
import { PropTypes } from 'prop-types';
import styles from './FilterButton.module.scss';

export default function FilterButton({ name, isChoosen, onChoose }) {
  const [hover, setHover] = useState(false);

  const handleMouseEnter = () => setHover(true);
  const handleMouseLeave = () => setHover(false);

  return (
    <button
      className={`${styles.filterButton} ${hover ? styles.mouseOn : ''} ${
        isChoosen ? styles.choosen : ''
      }`}
      type="button"
      name={name}
      onClick={onChoose}
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
    >
      {name}
    </button>
  );
}

FilterButton.propTypes = {
  name: PropTypes.string.isRequired,
  isChoosen: PropTypes.bool.isRequired,
  onChoose: PropTypes.func.isRequired,
};
