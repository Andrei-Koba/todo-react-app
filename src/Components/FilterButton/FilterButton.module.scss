.filterButton {
    border: 0;
    font-size: 15px;
    color: gray;
    background-color: transparent;
}

.mouseOn {
    color: black;
}

.choosen {
    font-weight: bold;
    color: rgb(62 124 217);
}