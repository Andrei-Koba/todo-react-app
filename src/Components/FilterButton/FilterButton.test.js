import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import FilterButton from './FilterButton';

describe('FilterButton', () => {
  it.each([true, false])('renders correctly, isChoosen="%p"', (isChoosen) => {
    const actual = render(
      <FilterButton name="testName" isChoosen={isChoosen} onChoose={jest.fn()} />,
    );
    expect(actual).toMatchSnapshot();
  });

  it('call  handleClick when choosen', async () => {
    const onChoose = jest.fn();
    const actual = render(<FilterButton name="testName" isChoosen={false} onChoose={onChoose} />);
    fireEvent.click(actual.getByRole('button'));
    expect(onChoose).toBeCalled();
  });

  it('add class on mouseEnter', async () => {
    const actual = render(<FilterButton name="testName" isChoosen={false} onChoose={jest.fn()} />);
    const button = actual.getByRole('button');
    fireEvent.mouseEnter(button);
    expect(button.classList.contains('mouseOn')).toBe(true);
    fireEvent.mouseLeave(button);
    expect(button.classList.contains('mouseOn')).toBe(false);
  });
});
