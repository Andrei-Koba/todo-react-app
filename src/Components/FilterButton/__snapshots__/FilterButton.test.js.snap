// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterButton renders correctly, isChoosen="false" 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="filterButton  "
        name="testName"
        type="button"
      >
        testName
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="filterButton  "
      name="testName"
      type="button"
    >
      testName
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FilterButton renders correctly, isChoosen="true" 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="filterButton  choosen"
        name="testName"
        type="button"
      >
        testName
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="filterButton  choosen"
      name="testName"
      type="button"
    >
      testName
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
